src/ast/ast.c

```diff
@@ -591,12 +611,12 @@ const char **AST_BuildReturnColumnNames
 	uint projection_count = cypher_ast_return_nprojections(return_clause);
 	const char **columns = array_new(const char *, projection_count);
 	for(uint i = 0; i < projection_count; i++) {
-		const cypher_astnode_t *projection = cypher_ast_return_get_projection(return_clause, i);
-		const cypher_astnode_t *ast_alias = cypher_ast_projection_get_alias(projection);
-		// If the projection was not aliased, the projection itself is an identifier.
-		if(ast_alias == NULL) ast_alias = cypher_ast_projection_get_expression(projection);
-		const char *alias = cypher_ast_identifier_get_name(ast_alias);
-		array_append(columns, alias);
+		const cypher_astnode_t *projection =
+			cypher_ast_return_get_projection(return_clause, i);
+		const char *alias = AST_GetProjectionAlias(projection);
+		if(alias != NULL) {
+			array_append(columns, alias);
+		}
```

Comment:
Seems like `AST_GetProjectionAlias` can't return `NULL` if my understanding is correct please remove this condition.

Reply:
OK. Condition removed in [23fa13d](https://github.com/RedisGraph/RedisGraph/pull/3107/commits/23fa13dd69f0528f3ca6b528f7b587ce54ba0569)